src/java.base/share/classes/jdk/internal/classfile/impl/CodeImpl.java

```diff
@@ -237,6 +237,10 @@ private void inflateLineNumbers() {
                 int pEnd = p + (nLn * 4);
                 for (; p < pEnd; p += 4) {
                     int startPc = classReader.readU2(p);
+                    if (startPc > codeLength) {
+                        throw new IllegalArgumentException(String.format("Line number out of range; start_pc=%d, codeLength=%d",
```

Comment:
It's the byte code index that is out of range, not the line number associated with it.

Reply:
Right, fixed the message, thanks!